Добавляет рецепт «Как опубликовать сайт прямо на GitHub» (#4743)
doka-guide/content: recipes/github-pages-static-site/index.md

---
title: "Как опубликовать сайт прямо на GitHub"
description: "Вы можете опубликовать свой сайт прямо на GitHub."
authors:
  - igsekor
related:
  - tools/version-control
  - tools/git-cli
  - tools/github-actions
tags:
  - article
---

## Задача

> Как опубликовать сайт в виде набора HTML страниц для тестирования или постоянного размещения прямо на GitHub. Например, такой сайт может стать отличным портфолио, хостинг и доменное имя третьего уровня для которого будут бесплатными.

Reviewer: ```suggestion
Как опубликовать сайт в виде набора HTML-страниц для тестирования или постоянного размещения прямо на GitHub. Например, такой сайт может стать отличным портфолио. Для него будут бесплатными хостинг и доменное имя третьего уровня.
```

## Готовое решение

> На GitHub есть для этого специальный инструмент — GitHub Pages, — который позволяет публиковать сайты, используя в качестве хостинга сам сервис. На сегодняшний день существуют два способа использования этого инструмента:

Reviewer: ```suggestion
На GitHub для этого есть специальный инструмент — GitHub Pages. С помощью него публикуют сайты, используя в качестве хостинга сам сервис. Есть два способа использования этого инструмента:
```

> 1. Собранный сайт помещается в отдельную папку в выбранной ветке репозитория, GitHub просто отображает эту папку как корневую для сайта. Такой подход можно использовать для показа уже собранного статического сайта в виде набора файлов в форматах, пригодных для отображения в браузере напрямую.
> 2. Статический сайт сначала собирается с помощью средств автоматизации GitHub Actions (об этом инструменте вы можете почитать в статье «[GitHub Actions](https://doka.guide/tools/github-actions/)») и публикуется также как в первом случае.

Reviewer: ```suggestion
1. Собранный сайт помещается в отдельную папку в выбранной ветке репозитория, GitHub просто отображает эту папку как корневую для сайта. Такой подход используют для показа уже собранного статического сайта в виде набора файлов в форматах, которые отображаются в браузере напрямую.
1. Статический сайт сначала собирается с помощью средств автоматизации GitHub Actions и публикуется также как в первом случае.

Про GitHub Actions можете почитать в статье «[GitHub Actions](/tools/github-actions/)».
```

> Второй способ мы рассматривать не будем, поскольку он находится на этапе бета-тестирования. Настройки для этого режима:

Reviewer: ```suggestion
Второй способ не будем рассматривать, так как он находится на этапе бета-тестирования. В этом режиме можно выбрать уже готовые решения, например, GitHub Pages Jekyll и Static HTML, а можно настроить свой воркфлоу.
```

> ![Работа GitHub Pages с помощью GitHub Actions (экспериментальный способ). Описание выше.](images/github-pages-actions-setup.png)

Reviewer: ```suggestion
![Работа GitHub Pages с помощью GitHub Actions.](images/github-pages-actions-setup.png)
```

> Первый способ рассмотрим подробно:

Reviewer: ```suggestion
Рассмотрим подробно первый способ.
```

> 1. Необходимо зайти на вкладку настроек репозитория «Settings» и перейти в раздел «Pages».

Reviewer: ```suggestion
На странице с настройками своего профиля откройте вкладку «Settings» и выберите в боковой навигации пункт «Pages» из раздела «Code and automation».
```

> ![Настройка GitHub Pages. Описание выше.](images/github-pages-branch-setup.png)

Reviewer: ```suggestion
![Настройки профиля и опция GitHub Pages.](images/github-pages-branch-setup.png)
```

> 2. В пункте «Source» нужно выбрать способ «Deploy from a branch».

Reviewer: ```suggestion
В развернувшемся содержимом вкладки найдите раздел с заголовком «Build and deployment». Раскройте выпадающий список «Source» рядом с заголовком и выберите способ «Deploy from a branch».
```

> ![Выбор способа работы GitHub Pages. Описание выше.](images/deploy-from-branch.png)

Reviewer: ```suggestion
![Выбор способа работы GitHub Pages.](images/deploy-from-branch.png)
```

> 3. В пункте «Branch» нужно выбрать нужную ветку репозитория и затем нужную папку с сайтом.

Reviewer: ```suggestion
В пункте «Branch» в содержимом той же вкладки выберите нужную ветку репозитория. Для этого введите её название в поле с подписью «Select brunch» или выберите одну из веток в выпадающем списке к полю. Затем выберите нужную папку с сайтом. Для этого нужно раскрыть соседний выпадающий список. Наконец, нажмите кнопку «Save».

> ![Выбор ветки для публикации сайта. Описание выше.](images/choose-the-branch.png)

Reviewer: ```suggestion
![Выбор ветки для публикации сайта.](images/choose-the-branch.png)
```

> ![Выбор папки для публикации сайта. Описание выше.](images/chosse-folder-from-branch.png)

Reviewer: ```suggestion
![Выбор папки для публикации.](images/chosse-folder-from-branch.png)
```

> После этого, инструмент GitHub Pages должен стать активным в репозитории, и по соответствующему адресу `<имя пользователя>.github.io/<имя репозитория>` должен стать доступным статический сайт из выбранной папки.

Reviewer: ```suggestion
Теперь инструмент GitHub Pages стал активным. По соответствующему адресу `<имя пользователя>.github.io/<имя репозитория>` через несколько минут станет доступным статический сайт из выбранной папки.
```
